src/connection.c

```diff
@@ -503,6 +508,52 @@ free_resolv_cb_data(struct resolv_cb_data *cb_data) {
 }
 
 static void
+send_proxy_header(int clientfd, const struct Connection *con) {
```

Comment:
The reason I used `getsockname` rather than `con->listener->address` is because this returns the correct local address in case you have bound the socket to a "wildcard" address such as 0.0.0.0 or ::.

I've found a way to send the header before any client data is pushed, and after the backend has been resolved. I'll update this PR soon.